src/remat.jl

```diff
@@ -33,48 +33,46 @@ Combine multiple ReMat with the same grouping variable into a single object.
 amalgamate(reterms::Vector{ReMat{T,S} where S}) where {T} = _amalgamate(reterms,T)
 # constant S
 amalgamate(reterms::Vector{ReMat{T,S}}) where {T,S} = _amalgamate(reterms,T)
+
 function _amalgamate(reterms::Vector, T::Type)
-    if length(reterms) == 1
-        return first(reterms)
+    factordict = Dict{Symbol, Vector{Int}}()
+    for (i, rt) in enumerate(reterms)
+        push!(get!(factordict, rt.trm.sym, Int[]), i)
```

Comment:
This is really nice. It took me a minute to think about why it isn't dependent on the original ordering of RE terms (e.g., whether it was sensitive an `item` term followed by a `subject` term followed by an `item` term).